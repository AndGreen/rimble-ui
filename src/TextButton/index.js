import React from 'react'
import styled from 'styled-components'
import {
  tint,
  shade
} from 'polished'

import Button from '../Button'

const TextButton = styled(Button)`
  & {
    box-shadow: none;
  }
  &:hover {
    color: ${props => tint(0.2, props.theme.colors.primary)};
    background-color: transparent;
    text-decoration: underline;
    box-shadow: none;
  }
  &:active {
    color: ${props => shade(0.2, props.theme.colors.primary)};
    text-decoration: none;
    background-color: transparent;
  }
`

TextButton.defaultProps = {
  fontSize: 'inherit',
  fontWeight: '600',
  color: 'primary',
  bg: 'transparent',
  height: '48px',
  border: 'none',
  px: 4,
  py: 0
}

TextButton.displayName = 'TextButton'

export default TextButton
