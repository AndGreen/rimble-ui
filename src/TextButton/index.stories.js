import { storiesOf } from '@storybook/react'

import React from 'react'

import TextButton from '../TextButton'

storiesOf('Button/TextButton', module)
  .add('default', () => (
    <div>
      <TextButton size='small'>Small TextButton</TextButton>
      <TextButton size='medium'>Medium TextButton</TextButton>
      <TextButton size='large'>Large TextButton</TextButton>
    </div>
  ))
