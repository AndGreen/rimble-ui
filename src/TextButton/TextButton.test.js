import TextButton from './'

describe('TextButton component sanity', () => {
  it('has name', () => {
    expect(TextButton.displayName).toBe('TextButton')
  })

  it('matches default snapshot', () => {
    const component = render(<TextButton />)
    expect(component).toMatchSnapshot()
  })
})
