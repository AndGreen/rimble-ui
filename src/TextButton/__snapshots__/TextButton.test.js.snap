// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton component sanity matches default snapshot 1`] = `
.c2 {
  padding-left: 32px;
  padding-right: 32px;
  padding-top: 0px;
  padding-bottom: 0px;
  color: #4E3FCE;
  background-color: transparent;
  font-size: inherit;
  height: 48px;
  border: none;
  text-align: left;
}

.c2 {
  box-sizing: border-box;
}

.c1 {
  font-weight: 600;
  font-size: 1rem;
  min-width: 3rem;
}

.c1 {
  cursor: pointer;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: 1;
  overflow: hidden;
  white-space: nowrap;
  -webkit-font-smoothing: antialiased;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
  background-color: #4E3FCE;
}

.c1:hover {
  background-color: #5f52d2;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.2);
}

.c1:active {
  background-color: #4638b9;
  box-shadow: none;
}

.c1:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

.c1 > svg {
  display: block;
}

.c1 > svg:first-child {
  margin-left: -1rem;
  margin-right: .5rem;
}

.c1 > svg:last-child {
  margin-left: .5rem;
  margin-right: -1rem;
}

.c1 > svg:first-child:last-child {
  margin: 0;
}

.c1 {
  background-color: transparent;
  box-shadow: none;
}

.c1:hover {
  color: #7165d7;
  background-color: transparent;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  box-shadow: none;
}

.c1:active {
  color: #3e32a4;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
}

.c0 {
  font-size: 1em;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  color: #3F3D4B;
}

<div
  class="c0"
>
  <button
    class="c1 c2"
    color="primary"
    font-size="inherit"
    font-weight="600"
    height="48px"
    type="button"
  >
    
  </button>
</div>
`;
